coral-common/src/main/java/com/linkedin/coral/common/TypeConverter.java

```diff
@@ -141,11 +141,16 @@ public static RelDataType convert(StructTypeInfo structType, final RelDataTypeFa
   }
 
   // Mimic the StructTypeInfo conversion to convert a UnionTypeInfo to the corresponding RelDataType
+  // The schema of output Struct conforms to https://github.com/trinodb/trino/pull/3483
   public static RelDataType convert(UnionTypeInfo unionType, RelDataTypeFactory dtFactory) {
     List<RelDataType> fTypes = unionType.getAllUnionObjectTypeInfos().stream()
         .map(typeInfo -> convert(typeInfo, dtFactory)).collect(Collectors.toList());
-    List<String> fNames = IntStream.range(0, unionType.getAllUnionObjectTypeInfos().size()).mapToObj(i -> "tag_" + i)
+    List<String> fNames = IntStream.range(0, unionType.getAllUnionObjectTypeInfos().size()).mapToObj(i -> "field" + i)
         .collect(Collectors.toList());
+    if (fNames.size() > 0) {
```

Comment:
I think we don't need this if statement since an empty union is a wrong schema, to begin with, maybe we could add a `Preconditions.checkState` to check it.

Reply:
Seems that here https://github.com/trinodb/trino/pull/3483/files# the size is not checked. Does the Avro or ORC standard say anything about this?

Reply:
ORC spec doesn't mention it while I do see `uniontype<>` happening in production. Regardless, this check is there to keep parity with the original (or we could end up with an additional `tag` field).

Reply:
So if using the current code, `uniontype<>` ends up in an empty struct? I recall there are some issues with empty struct somewhere, either in Iceberg or in Spark. So I'm not sure if this whole corner case will work or not.
My point is if we explicitly announce this special case's behavior is undefined, then users have the responsibility to create the correct schema, it's a tradeoff between us and the users, I'm willing to accept either way.

Reply:
Our references should be: 1- what the standard says (again for Avro or ORC) 2- what the Trino transformation does. Hopefully all of them align. If not, we can discuss how to move forward.

Reply:
Another special case is union with only a single type `uniontype<[type]>`. We should consider whether we want to support it and make sure its semantics is consistent across Trino and Iceberg implementation

Reply:
There aren't any spec that I could find regarding this point in either Avro or ORC, and looks like Trino doesn't check the size either. 
Taking a step back, I also don't think Coral is in right position to gate such usage. So I am leaning towards let such case pass through (which means a check like `fNames.size > 0 ` is necessary since we see `struct<>` as a legit type). 

I am also OK to add a precondition check to fail the translation. But again, leaning towards the former option.

Reply:
+1 to the current approach

Reply:
Discussed offline. The objective is to match the Trino schema. We should remove the check from here since it is not used there.